packages/next/server/next-server.ts

```diff
@@ -696,6 +711,40 @@ export default class Server {
           }
         },
       },
+      {
+        match: route('/_next/flight/:path*'),
+        type: 'route',
+        name: '_next/flight catchall',
+        fn: async (req, res, params, _parsedUrl) => {
+          // Make sure to 404 for /_next/data/ itself and
+          // we also want to 404 if the buildId isn't correct
+          if (!params.path || params.path[0] !== this.buildId) {
+            await this.render404(req, res, _parsedUrl)
+            return {
+              finished: true,
+            }
+          }
+          // remove buildId from URL
+          params.path.shift()
+
+          // re-create page's pathname
+          let pathname = `/${params.path.join('/')}`
+          pathname = getRouteFromAssetPath(pathname, '.json')
+
+          const parsedUrl = parseUrl(pathname, true)
+
+          await this.render(
+            req,
+            res,
+            pathname,
+            { ..._parsedUrl.query, _nextFlightReq: '1' },
+            parsedUrl
+          )
+          return {
+            finished: true,
+          }
+        },
+      },
```

Comment:
This is fairly minor, but I was expecting that we'd probably just re-use the existing `/_next/data` catchall, because it already has most of the page data/html duality that we want. I guess one complexity is that those require the `*.json` extension now, and the RSC protocol isn't actually JSON...

Reply:
IIRC that `.json` extension can be ignored, but reusing `/data` sounds good to me. 👍